src/io/DIAmeterFeatureScaler.cpp

```diff
@@ -0,0 +1,114 @@
+
+//#include "MatchColumns.h"
+#include "DIAmeterFeatureScaler.h"
+#include "DelimitedFile.h"
+#include "carp.h"
+
+using namespace std;
+
+DIAmeterFeatureScaler::DIAmeterFeatureScaler(const char* file_name) {
+	fileReader_ = new DelimitedFileReader(file_name, true);
+    parseHeader();
+}
+
+DIAmeterFeatureScaler::~DIAmeterFeatureScaler() {
+	delete fileReader_;
+	for(std::vector<int>::iterator it = toscale_column_indices_.begin(); it != toscale_column_indices_.end(); ++it) {
+		delete toscale_match_values_[*it];
+	}
+}
+
+void DIAmeterFeatureScaler::writeScaledFile(const char* output_file_name) {
+	 fileReader_->reset();
+	 std::vector<std::string> output_vec;
+	 std::vector<std::string> column_names = fileReader_->getColumnNames();
+
+	 ofstream* output_file = create_stream_in_path(output_file_name, NULL, Params::GetBool("overwrite"));
+	 *output_file << StringUtils::Join(column_names, '\t').c_str() << endl;
+
+	 while (fileReader_->hasNext()) {
+		 output_vec.clear();
+		 for (int column_idx = 0; column_idx < column_names.size(); column_idx++) { output_vec.push_back(fileReader_->getString(column_idx)); }
+
+		 for (int idx = 0; idx < toscale_column_indices_.size(); idx++) {
+			 int curr_column_idx = toscale_column_indices_.at(idx);
```

Comment:
move the int and double declaration outside the loop so they are not made a bunch of times